source/Coop.Core/Server/Services/Sync/Handlers/PeerQueueOverloadedHandler.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Common.Network;
+using Common.Util;
+using Coop.Core.Server.Connections;
+using Coop.Core.Server.Services.Time.Handlers;
+using Coop.Core.Server.Services.Time.Messages;
+using GameInterface.Services.GameDebug.Messages;
+using GameInterface.Services.Heroes.Enum;
+using Serilog;
+using System;
+namespace Coop.Core.Server.Services.Sync.Handlers
+{
+    internal class PeerQueueOverloadedHandler : IHandler
+    {
+        private static readonly long POLL_INTERVAL = 500;
+
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+
+        private readonly TimeHandler timeHandler;
+        private readonly ClientRegistry clientRegistry;
+
+        private readonly ILogger logger;
+        private readonly Poller poller;
+
+        private TimeControlEnum originalSpeed;
+
+        private readonly object syncLock = new();
+
+        public PeerQueueOverloadedHandler(
+            IMessageBroker messageBroker, 
+            INetwork network,
+            ClientRegistry clientRegistry,
+            TimeHandler timeHandler
+        )
+        {
+            this.messageBroker = messageBroker;
+            this.network = network;
+            this.clientRegistry = clientRegistry;
+
+            this.timeHandler = timeHandler;
+
+            logger = LogManager.GetLogger<PeerQueueOverloaded>();
+
+            poller = new(Poll, TimeSpan.FromMilliseconds(POLL_INTERVAL));
+
+            messageBroker.Subscribe<PeerQueueOverloaded>(Handle);
+        }
+
+        private void Handle(MessagePayload<PeerQueueOverloaded> payload)
+        {
+            lock (syncLock)
+            {
+                if (clientRegistry.IsOverloaded(payload.What.NetPeer))
+                    return;
```

Comment:
This is probably fine but I would rather have this be isolated in this class. Create a List<NetPeer> overloadedClients and use that instead.

For this specific check

`overloadedClients.Contains(payload.What.NetPeer)`